shared/units/units.go

```diff
@@ -5,6 +5,19 @@ import (
 	"strconv"
 )
 
+func handleOverflow(val int64, mult int64) (int64, error) {
+	result := val * mult
+	if val == 0 || mult == 0 || val == 1 || mult == 1 {
```

Comment:
is there a case on which case mult can be 0?

Reply:
Not currently but if this is used anywhere else this check can't hurt.